### PR TITLE
initial mock feature file for staking and delegation fees

### LINKED ISSUE #564: Coded features review: on-chain rewards
I suspect this should be updated to point to this spec: https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0057-reward-functions.md#staking-and-delegation-required-for-

A lot of the work on the on chain rewards system is being sidestepped for [Sweetwater](https://github.com/vegaprotocol/specs-internal/blob/master/milestones/2.5-Sweetwater.md) - rather than testing 'on chain rewards' in total, we should verify that the one reward required for Sweetwater works as expected. The reward is outlined in the spec I put at the top. That reward is calculated as per https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0058-simple-POS-rewards.md

There are some other relevant specs:
- https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0058-simple-POS-rewards.md
- https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0056-rewards-overview.md
- https://github.com/vegaprotocol/specs-internal/blob/master/protocol/0055-on-chain-treasury.md

### PR DESCRIPTION
**Pull request by eyongegbe**
First stab on feature file for implementing integration tests for staking and delegation fees/rewards for validators and delegators.

Closes #563

### CODE REVIEW

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+#setup
+    Given the simple risk model named "simple-risk-model-1":
+      | long | short | max move up | min move down | probability of trading |
+      | 0.1  | 0.1   | 500         | 500           | 0.1                    |
+    And the fees configuration named "fees-config-1":
+      | maker fee | infrastructure fee |
+      | 0.0004    | 0.001              |
+    And the rewards configuration are set:
+      |reward fee|
+      |0.2       |
```
Comment (witgaw) [inline qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: Not sure what a reward fee is?

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    Given the following parties are active in the network:
+      | partyType      |PartyId|period|
+      | validator1     |1      |0     |   
+    And the parties deposit on asset's general account the following amount:
+      |party     |asset|amount|
+      |validator1|VEGA |500000|
+
+    When the staking rewards are calculated
+
+    Then the infrastructure fees bucket must contain fees for staking and delegation
```
Comment (witgaw) [inline qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: I don't think these are meant to go to the infrastructure fee bucket, rewards are a completely separate workflow, it's just that they are calculated in the same way.

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      | party      | asset | amount |
+      | validator1 | VEGA  | 500000 |
+      | delegator1 | VEGA  | 500000 |
+
+    When the staking rewards are calculated
+
+    Then the infrastructure fees bucket must contain fees for staking and delegation
+    And the infrastructure fees for staking and delegation should be:
+      | partyType      | partyId|feeType                | feeAmount | period |
+      | validator1     |   1    |staking and delegation | 21        | 0      |
```
Comment (witgaw) [inline qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: Where does the "21" come from?

**Review by witgaw — COMMENTED** [review]
Hey, I'm not too sure, on the details here (again, just based on my reading of the specs). Added some questions/comments below.

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock (hunk 1 above)
Comment (eyongegbe) [reply qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: The values are mock values. My thinking here was rewards are a percentage of the overall infrastructure fees. I suppose this is not the case?

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock (hunk 3 above)
Comment (eyongegbe) [reply qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: The amount is a mock argument/value

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock (hunk 1 above)
Comment (witgaw) [reply qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: I don't think so, they are a fraction of the reward balance. The term "fee" is just a bit confusing and it seems to refer to another fee type. In fact this entire feature file probably doesn't even need a market setup - these rewards are meant to work even in absence of any trading (as will be the case for the Sweetwater release).

File: qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock (hunk 3 above)
Comment (witgaw) [reply qa-scenarios/on-chain-rewards-staking-and-delegation.feature.mock]: I don't understand how it's meant to work though. The reward balance can certainly be mocked, but the distribution to parties (which I assume this is meant to represent) should be the calculated as per rules outlined in the spec.

**Comment by witgaw** [comment]
covered by https://github.com/vegaprotocol/vega/pull/4024

**Pull request closed without merging**